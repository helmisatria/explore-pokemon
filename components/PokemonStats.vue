<template>
  <ul class="space-y-4">
    <li
      v-for="(statistic, index) in stats"
      :key="`${index}-${statistic.stat.name}`"
      class="stats-item"
    >
      <figure>
        <figcaption class="text-sm">
          <span class="capitalize font-semibold">{{ statistic.stat.name }}</span>
          <span class="text-gray-500">({{ statistic.base_stat }})</span>
        </figcaption>
        <span class="mt-2 shadow-sm block relative w-full bg-white h-[6px] rounded-lg">
          <span
            class="block bg-green-400 h-[6px] rounded-lg"
            :style="`width: ${statistic.base_stat > 100 ? 100 : statistic.base_stat}%`"
          ></span>
        </span>
      </figure>
    </li>
  </ul>
</template>

<script>
export default {
  props: {
    stats: {
      type: Array,
      required: true,
    },
  },
}
</script>
