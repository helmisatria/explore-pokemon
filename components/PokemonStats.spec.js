import { createLocalVue, shallowMount } from '@vue/test-utils'
import { PiniaVuePlugin } from 'pinia'
import { createTestingPinia } from '@pinia/testing'
import PokemonStats from './PokemonStats.vue'

import pokemonDetailData from '~/fixtures/pokemon-detail.json'

describe('Component: PokemonStats', () => {
  let wrapper

  const localVue = createLocalVue()
  localVue.use(PiniaVuePlugin)

  const generateWrapper = () => {
    return shallowMount(PokemonStats, {
      localVue,
      pinia: createTestingPinia(),
      propsData: {
        stats: pokemonDetailData.species[0].pokemons[0].stats,
      },
    })
  }

  beforeEach(() => {
    wrapper = generateWrapper()
  })

  it('should mount component properly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should display stats item properly', () => {
    expect(wrapper.findAll('.stats-item').length).toBe(
      pokemonDetailData.species[0].pokemons[0].stats.length
    )
  })
})
