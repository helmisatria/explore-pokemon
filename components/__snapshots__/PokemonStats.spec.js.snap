// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PokemonStats should mount component properly 1`] = `
<ul
  class="space-y-4"
>
  <li
    class="stats-item"
  >
    <figure>
      <figcaption
        class="text-sm"
      >
        <span
          class="capitalize font-semibold"
        >
          hp
        </span>
         
        <span
          class="text-gray-500"
        >
          (39)
        </span>
      </figcaption>
       
      <span
        class="mt-2 shadow-sm block relative w-full bg-white h-[6px] rounded-lg"
      >
        <span
          class="block bg-green-400 h-[6px] rounded-lg"
          style="width: 39%;"
        />
      </span>
    </figure>
  </li>
  <li
    class="stats-item"
  >
    <figure>
      <figcaption
        class="text-sm"
      >
        <span
          class="capitalize font-semibold"
        >
          attack
        </span>
         
        <span
          class="text-gray-500"
        >
          (52)
        </span>
      </figcaption>
       
      <span
        class="mt-2 shadow-sm block relative w-full bg-white h-[6px] rounded-lg"
      >
        <span
          class="block bg-green-400 h-[6px] rounded-lg"
          style="width: 52%;"
        />
      </span>
    </figure>
  </li>
  <li
    class="stats-item"
  >
    <figure>
      <figcaption
        class="text-sm"
      >
        <span
          class="capitalize font-semibold"
        >
          defense
        </span>
         
        <span
          class="text-gray-500"
        >
          (43)
        </span>
      </figcaption>
       
      <span
        class="mt-2 shadow-sm block relative w-full bg-white h-[6px] rounded-lg"
      >
        <span
          class="block bg-green-400 h-[6px] rounded-lg"
          style="width: 43%;"
        />
      </span>
    </figure>
  </li>
  <li
    class="stats-item"
  >
    <figure>
      <figcaption
        class="text-sm"
      >
        <span
          class="capitalize font-semibold"
        >
          special-attack
        </span>
         
        <span
          class="text-gray-500"
        >
          (60)
        </span>
      </figcaption>
       
      <span
        class="mt-2 shadow-sm block relative w-full bg-white h-[6px] rounded-lg"
      >
        <span
          class="block bg-green-400 h-[6px] rounded-lg"
          style="width: 60%;"
        />
      </span>
    </figure>
  </li>
  <li
    class="stats-item"
  >
    <figure>
      <figcaption
        class="text-sm"
      >
        <span
          class="capitalize font-semibold"
        >
          special-defense
        </span>
         
        <span
          class="text-gray-500"
        >
          (50)
        </span>
      </figcaption>
       
      <span
        class="mt-2 shadow-sm block relative w-full bg-white h-[6px] rounded-lg"
      >
        <span
          class="block bg-green-400 h-[6px] rounded-lg"
          style="width: 50%;"
        />
      </span>
    </figure>
  </li>
  <li
    class="stats-item"
  >
    <figure>
      <figcaption
        class="text-sm"
      >
        <span
          class="capitalize font-semibold"
        >
          speed
        </span>
         
        <span
          class="text-gray-500"
        >
          (65)
        </span>
      </figcaption>
       
      <span
        class="mt-2 shadow-sm block relative w-full bg-white h-[6px] rounded-lg"
      >
        <span
          class="block bg-green-400 h-[6px] rounded-lg"
          style="width: 65%;"
        />
      </span>
    </figure>
  </li>
</ul>
`;
